pkg/windows/calico.go

```diff
@@ -60,24 +62,28 @@ func (c *Calico) Setup(ctx context.Context, dataDir string, nodeConfig *daemonco
 		return nil, err
 	}
 
-	secrets, err := client.CoreV1().Secrets(CalicoSystemNamespace).List(ctx, metav1.ListOptions{})
+	sa, err := serviceaccount.GetOrCreateServiceAccount(client.CoreV1(), CalicoSystemNamespace, calicoNode)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, secret := range secrets.Items {
-		if !isCalicoNodeToken(&secret) {
-			continue
-		}
-		value, ok := secret.Data["token"]
-		if !ok {
-			continue
-		}
-		calicoKubeConfig.Token = string(value)
+	tokenRequest := authentication.TokenRequest{
+		TypeMeta:   sa.TypeMeta,
+		ObjectMeta: sa.ObjectMeta,
+		Spec: authentication.TokenRequestSpec{
+			BoundObjectRef: &authentication.BoundObjectReference{
+				Kind:       sa.Kind,
+				APIVersion: sa.APIVersion,
+				Name:       CalicoSystemNamespace,
+			},
+		},
 	}
-	if calicoKubeConfig.Token == "" {
-		return nil, errors.New("could not retrieve calico node token from the cluster")
+	requestStatus := tokenRequest.Status
+
+	if requestStatus.Token == "" {
```

Comment:
You appear to be just creating a TokenRequest object and then checking its Token field. You need to actually request the apiserver create the token for you by creating a client and calling `CreateToken()`. This will give you an error to check/wrap, and filled TokenRequest to pull the token from:

https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/core/v1#ServiceAccountInterface

Reply:
good catch. latest commit should have this addressed